app/services/bulk_hosts_manager.rb

```diff
@@ -0,0 +1,35 @@
+class BulkHostsManager
+  def initialize(hosts:)
+    @hosts = hosts
+  end
+
+  def build(reboot: false)
+    # returns missed hosts
+    @hosts.select do |host|
+      success = true
+      begin
+        host.built(false) if host.build? && host.token_expired?
+        host.setBuild
+        host.power.reset if host.supports_power_and_running? && reboot
```

Comment:
seems like this may save some compute

```suggestion
        host.power.reset if reboot && host.supports_power_and_running?
```